browsers/duckduckgo.safariextension/js/background.js

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (C) 2012, 2016 DuckDuckGo, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+var debugRequest = false;
+var trackers = require('trackers');
+var utils = require('utils');
+var settings = require('settings');
+var https = require('https');
+let browser = 'safari'
+
+function Background() {
+  $this = this;
+}
+
+var background = new Background();
+
+var handleMessage = function (message) {
+    if (message.name === 'canLoad') {
+        return onBeforeRequest(message)
+    }
+}
+
+// when the extension first installs, check for any existing tabs
+// and create a background tab so we can show the correct site in the popup
+function onInstalled () {
+    if(!localStorage['installed']) {
+        safari.application.browserWindows.forEach((safariWindow) => {
+            safariWindow.tabs.forEach((safariTab) => {
+                // create a tab id and store in safari tab
+                safariTab.ddgTabId = Math.floor(Math.random() * (1000 - 10 + 1)) + 10
```

Comment:
Should we do anything to prevent collisions? probably an edge case, but with < 1000 possibilities and people that open tons of tabs, I could see it happening. Where did the 1000 - 10 + 1 + 10 come from?

Reply:
I'm open to suggestions. I just made it up.

Reply:
I could always make the range larger and check for an existing ID.

Reply:
yea, either of those seem good enough. I'd just add a few 0's to the number if checking for id isn't easy.

Reply:
Sounds good. I'll update it.